ui/src/app/services/error-handling.service.ts

```diff
@@ -93,4 +90,18 @@ export class ErrorHandlingService {
       });
     });
   }
+
+  // convert error response from API to ErrorResponse object
+  public convertAPIError (e: any) {
+    if (e._body != null && JSON.parse(e._body) != null) {
+      const convertedError: ErrorResponse = {
+        'errorClassName': JSON.parse(e._body).errorClassName || '',
```

Comment:
Can we store the value of JSON.parse(e._body) here in order to avoid needing to parse a bunch of times?